wandb/sdk/launch/registry/azure_container_registry.py

```diff
@@ -0,0 +1,81 @@
+"""Implementation of AzureContainerRegistry class."""
+import re
+from typing import Tuple
+
+from ..environment.abstract import AbstractEnvironment
+from ..environment.azure_environment import AzureEnvironment
+from ..errors import LaunchError
+from .abstract import AbstractRegistry
+
+
+class AzureContainerRegistry(AbstractRegistry):
+    """Helper for accessing Azure Container Registry resources."""
+
+    def __init__(
+        self,
+        environment: AzureEnvironment,
+        uri: str,
+        verify: bool = True,
+    ):
+        """Initialize an AzureContainerRegistry."""
+        self.environment = environment
+        self.uri = uri
+        if verify:
+            self.verify()
+
+    @classmethod
+    def from_config(
+        cls, config: dict, environment: AbstractEnvironment, verify: bool = True
+    ) -> "AzureContainerRegistry":
+        """Create an AzureContainerRegistry from a config dict."""
+        if not isinstance(environment, AzureEnvironment):
+            raise LaunchError(
+                "AzureContainerRegistry requires an AzureEnvironment to be passed in."
+            )
+        uri = config.get("uri")
+        if uri is None:
+            raise LaunchError(
+                "Please specify a registry name to use under the registry.uri."
+            )
+        return cls(
+            uri=uri,
+            environment=environment,
+            verify=verify,
+        )
+
+    def get_username_password(self) -> Tuple[str, str]:
+        """Get username and password for container registry."""
+        raise NotImplementedError
+
+    def check_image_exists(self, image_uri: str) -> bool:
+        """Check if image exists in container registry.
+
+        WARNING: This is not implemented for Azure Container Registry and will
```

Comment:
Is this a TODO? Seems like this will make our Azure build ops pretty ineffecient

Reply:
Okay I'm gonna do this